Palindrome.java

```diff
@@ -1,30 +1,105 @@
-/**
- * A Palindrome is a sequence of characters which reads the same backward as forward.
- * Examples: anna, civic, kayak, level, madam,...
- * We will consider only numbers that can be called Palindrome. For example: 1001, 242, 99988999,...
- * The program will ask the user to enter a number (int), between 0 and 1,000,000,000.
- * It will check the input and proceed only when the input is finally correct.
- * If the input is NOT a Palindrome it will print a suitable message and finish.
- * Else, it will print the following (see examples next):
- * 1: the length of the number
- * 2: the first part of the Palindrome with spaces between the characters
- * 3: the second part of the Palindrome with spaces between the characters
- * NOTES:
- * 1. You must use ONLY things we learned (don't depend on string manipulations for example)
- * 2. Leading zeros don't count. For example, 00100 is NOT a Palindrome because the real number is 100
- *
- * Examples:
- * 1. Input: 13531
- *    Output: size = 5
- *    First part = 1 3 5      (with a space between each 2 characters)
- *    Second part = 5 3 1     (with a space between each 2 characters)
- * 2. Input: 98433489
- *    Output: size = 8
- *    First part = 9 8 4 3      (with a space between each 2 characters)
- *    Second part = 3 4 8 9     (with a space between each 2 characters)
- */
+import java.util.Scanner;
+import java.util.Arrays;
+
 public class Palindrome {
-    public static void main(String[] args) {
-        // Your code goes here :)
-    }
+	public static void main(String[] args) {
+
+		//Asks for a number
+		Scanner in = new Scanner(System.in);
+		System.out.println("please enter a number between 0 and 1,000,000,000");
+		int n = in.nextInt();
+
+		//Checks numer's validity
+		while (n < 0 || n > 1000000000) {
+			System.out.println("please enter a valid number");
+			n = in.nextInt();
+
+		}
+
+		in.close();
+
+		//Calculates array's length
+		int arrLength;
+
+		if (n < 100000) {

```

Comment:
the length of the array can be calculated via "%" and "/" operators. This way, if the upper bound is set to something else (like 10000 for example) you don't have to do many changes.